File: AN_2/SEM_2/MPP/java/lab4/src/main/java/ro/mpp/labfx/service/ParticipantService.java
package ro.mpp.labfx.service;

import ro.mpp.labfx.domain.Echipa;
import ro.mpp.labfx.domain.Participant;
import ro.mpp.labfx.domain.ParticipantDTO;
import ro.mpp.labfx.repository.ParticipantRepository;
import java.util.List;

public class ParticipantService {
    private ParticipantRepository participantRepository;

    public ParticipantService(ParticipantRepository participantRepository) {
        this.participantRepository = participantRepository;
    }

    public List<Participant> findAll() {
        List<Participant> participanti = participantRepository.findAll();
        return participanti;
    }

    public List<ParticipantDTO> getParticipantiByEchipa(Echipa echipa) {
        List<ParticipantDTO> participanti = participantRepository.findByEchipa(echipa);
        return participanti;
    }


    public void inscriereParticipant(Participant participant) {
        participantRepository.adauga(participant);
    }

    public Participant findbyNume(Participant participant) {
        return participantRepository.findbynume(participant.getNume());
    }
}

File: AN_2/SEM_2/MPP/c#/lab4/lab4/service/ParticipantService.cs
using lab4.domain;
using log4net;

namespace lab4.service;

public class ParticipantService(IParticipantRepository participantRepo, IEchipaRepository echipaRepo, Validare validare)
{
    private static readonly ILog Logger = LogManager.GetLogger(typeof(ParticipantService));

    public void InscriereParticipant(Participant participant)
    { 
        validare.ValideazaParticipant(participant.GetNume(), participant.GetCnp(), participant.GetEchipa().GetNume());
        participantRepo.Adauga(participant);
    }

    public List<Participant> FindAllParticipanti()
    {
        Logger.Debug("Incep cautarea tuturor participantilor.");
        return participantRepo.FindAll();
    }
    
    public List<ParticipantDTO> CautaParticipantiDupaEchipa(string numeEchipa)
    {
        Logger.Debug($"Caut participantii pentru echipa {numeEchipa}");
        validare.ValideazaEchipa(numeEchipa);
        Echipa echipa = echipaRepo.FindByNume(numeEchipa);
        return participantRepo.FindByEchipa(echipa);
    }
}